choreolib/src/main/native/cpp/choreo/lib/ChoreoSwerveCommand.cpp

```diff
@@ -0,0 +1,45 @@
+#include "choreo/lib/ChoreoSwerveCommand.h"
+
+#include <frc/DriverStation.h>
+
+using namespace choreolib;
+
+ChoreoSwerveCommand::ChoreoSwerveCommand(ChoreoTrajectory trajectory,
+		std::function<frc::Pose2d()> poseSupplier,
+		ChoreoControllerFunction controller,
+		std::function<void(frc::ChassisSpeeds)> outputChassisSpeeds,
+		bool useAllianceColor, frc2::Requirements requirements) : m_traj(
+		trajectory), m_pose(poseSupplier), m_controller(controller), m_outputChassisSpeeds(
+		outputChassisSpeeds), m_useAlliance(useAllianceColor) {
+	AddRequirements(requirements);
+}
+
+// Called when the command is initially scheduled.
+void ChoreoSwerveCommand::Initialize() {
+	m_timer.Restart();
+}
+
+// Called repeatedly when this Command is scheduled to run
+void ChoreoSwerveCommand::Execute() {
+	bool mirror = false;
+	if (m_useAlliance) {
+		std::optional < frc::DriverStation::Alliance > alliance =
+				frc::DriverStation::GetAlliance();
+		mirror = alliance.has_value()
+				&& alliance.value() == frc::DriverStation::Alliance::kRed;
+	}
+	units::second_t currentTrajTime = m_timer.Get();
+	m_outputChassisSpeeds(
+			m_controller(m_pose(), m_traj.Sample(currentTrajTime, mirror)));
```

Comment:
The example code for alliance color at https://docs.wpilib.org/en/latest/docs/software/basic-programming/alliancecolor.html is cleaner. Fetching this for every execute call seems like a bad idea. Maybe it's better to let the user handle this.